javascript/011-custom-signs/custom-signs.js

```diff
@@ -0,0 +1,64 @@
+// @ts-check
+
+// The line above enables type checking for this file. Various IDEs interpret
+// the @ts-check directive. It will give you helpful autocompletion when
+// implementing this exercise.
+
+/**
+ * Build a sign that includes both of the parameters.
+ *
+ * @param {string} occasion
+ * @param {string} name
+ *
+ * @returns {string} template string combining both parameters
+ */
+
+export function buildSign(occasion, name) {
+  return `Happy ${occasion} ${name}!`;
+}
+
+/**
+ * Build a birthday sign that conditionally formats the return string.
+ *
+ * @param {number} age
+ *
+ * @returns {string} template string based on age
+ */
+
+export function buildBirthdaySign(age) {
+  const pivotAge = 50;
+  const determineMaturity = age < pivotAge ? 'young' : 'mature';
+
+  return `Happy Birthday! What a ${determineMaturity} fellow you are.`
+}
+
+/**
+ * Build a graduation sign that includes multiple lines.
+ *
+ * @param {string} name
+ * @param {number} year
+ *
+ * @returns {string} multi-line template string
+ */
+
+export function graduationFor(name, year) {
+ return `Congratulations ${name}!\nClass of ${year}`;
+}
+
+/**
+ * Determine cost based on each character of sign parameter that builds
+ * the template string that includes the currency parameter.
+ *
+ * @param {string} sign
+ * @param {string} currency
+ *
+ * @returns {string} cost to create the sign
+ */
+
+export function costOf(sign, currency) {
+  const baseSignCost = 20;
+  const letterCost = 2;
+  const calculatedPrice = baseSignCost + (sign.trim().length * letterCost);
```

Comment:
Re-read the README about how whitespaces should be handled. Add a test for that.